.pre-commit-config.yaml

```diff
@@ -0,0 +1,27 @@
+# Copyright (c) 2024, NVIDIA CORPORATION.
+ci:
+    autofix_commit_msg: |
+      [pre-commit.ci] auto code formatting
+    autofix_prs: false
+    autoupdate_branch: ''
+    autoupdate_commit_msg: '[pre-commit.ci] pre-commit autoupdate'
+    autoupdate_schedule: quarterly
```

Comment:
Note that if we enable pre-commit.ci, the autoupdate schedule for pre-commit.ci cannot be turned off. For cuCollections, they want to keep their clang-format version fixed at a certain value. CCCL would be the same way. The pre-commit.ci bot will open PRs like this one on a quarterly basis (this is the slowest setting allowed), and those PRs can be closed or modified to keep the clang-format version at the desired value. https://github.com/NVIDIA/cuCollections/pull/450